vscode-dotnet-sdk-extension/src/test/functional/DotnetCoreAcquisitionExtension.test.ts

```diff
@@ -225,14 +233,117 @@ suite('DotnetCoreAcquisitionExtension End to End', function() {
+    assert.strictEqual(await resolver.getFullySpecifiedVersion(), newestVersion);
+
+    resolver = new GlobalInstallerResolver(mockExtensionContext, eventStream, featureBandOnlyVersion);
+    resolver.customWebRequestWorker = webWorker;
+    assert.strictEqual(await resolver.getFullySpecifiedVersion(), newestBandedVersion);
+
+    if(os.arch() === 'x64')
+    {
+      // We check this only on x64 because that matches the build machines and we dont want to duplicate architecture mapping logic
+      if(os.platform() === 'win32')
+      {
+        const expectedWinInstallerUrl = 'https://download.visualstudio.microsoft.com/download/pr/dotnet-sdk-6.0.311-win-x64.exe';
+        assert.strictEqual(await resolver.getInstallerUrl(), expectedWinInstallerUrl);
+      }
+      else if(os.platform() === 'darwin')
+      {
+        const expectedMacInstallerUrl = 'https://download.visualstudio.microsoft.com/download/pr/dotnet-sdk-6.0.311-osx-x64.pkg';
+        assert.strictEqual(await resolver.getInstallerUrl(), expectedMacInstallerUrl);
+      }
+    }
+
+    resolver = new GlobalInstallerResolver(mockExtensionContext, eventStream, fullVersion);
+    resolver.customWebRequestWorker = webWorker;
+    assert.strictEqual(await resolver.getFullySpecifiedVersion(), fullVersion);
+  }).timeout(standardTimeoutTime);
+
+
+  test('Install Globally E2E (Requires Admin)', async () => {
+    // We only test if the process is running under ADMIN because non-admin requires user-intervention.
+    if(new FileUtilities().isElevated())
+    {
+      const originalPath = process.env.PATH;
+      const sdkVersion = '7.0.103';
+      const context : IDotnetAcquireContext = { version: sdkVersion, requestingExtensionId: 'ms-dotnettools.sample-extension', installType: 'global' };
+
+      // We cannot use the describe pattern to restore the environment variables using vscodes extension testing infrastructure.
+      // So we must set and unset it ourselves, which isn't ideal as this variable could remain.
+      let result : IDotnetAcquireResult;
+      let error : any;
+      let pathAfterInstall;
+
+      // We cannot test much as we don't want to leave global installs on devboxes. But we do want to make sure the e-2-e goes through the right path. Vendors can test the rest.
+      // So we have this environment variable that tells us to stop before running any real install.
+      process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH = 'true';
+      try
+      {
+        result = await vscode.commands.executeCommand<IDotnetAcquireResult>('dotnet-sdk.acquire', context);
+      }
+      catch(err)
+      {
+        error = err;
+      }
+      finally
+      {
+        pathAfterInstall = process.env.PATH;
+        process.env.VSCODE_DOTNET_GLOBAL_INSTALL_FAKE_PATH = undefined;
+        process.env.PATH = originalPath;
+
+        if(error)
+        {
+          throw(new Error(`The test failed to run the acquire command successfully. Error: ${error}`));
```

Comment:
nit: this flow seems somewhat odd to me- is there any reason to save the error as a var and rethrow here? Can we not just throw directly in the catch statement?

Reply:
This is so the test can still fail, but the environment variables can be restored whether the test gets to the end of the try block or the catch block, though I do agree it is somewhat clunky